chain/network/src/routing.rs

```diff
@@ -786,16 +785,18 @@ impl RoutingTable {
         }
 
         let component_nonce = index_to_bytes(component_nonce);
-        let mut edges_to_remove = vec![];
-
-        self.edges_info.retain(|(peer0, peer1), edge| {
-            if to_save.contains(peer0) || to_save.contains(peer1) {
-                edges_to_remove.push(edge.clone());
-                false
-            } else {
-                true
-            }
-        });
+        let edges_to_remove = self
```

Comment:
the fact that this bug existed, means that we didn't have any test covering that.

When you do a hotfix like that (unless it is super-emergency etc) - you should also add a test case, that would be "failing" before this fix and then passing afterwards.


in our case, raw_graph is used to calculate distance, so a test that removes the edge and makes sure that the calculated distance is changed accordingly would be good (at the same time, it is strange that no other test has detected this problem -- do we know why ?

Reply:
@mm-near I already added check for that issue in the tests. Take a look at:
```
        let active_edges = on_memory
            .iter()
            .filter_map(|x| if x.2 == EdgeType::Added { Some(1) } else { None })
            .count();

        assert_eq!(active_edges, self.routing_table.raw_graph.total_active_edges as usize);**
```

Reply:
ah, right. (I have to get used having "tests" in the same file as production code).

Having said that -- do we know why no other test has caught it ? Do we have a one that removes the edge and recalculates the distances ?

Reply:
Yes, we have tests for that. We didn't check whenever ` self.routing_table.raw_grap` has correct number of edges.

Reply:
checking whether edges in graph (which is internal structure) are matching edge_list (another structure) is testing the "implementation" not the feature.

We should add (in separate PR then) - a test that does the edge removal & distance recalculation -- as this is the one that would have failed without this PR.